### ISSUE TITLE
Failure to connect crashes Node, exception cannot be caught, connect blocks forever

### ISSUE THREAD

**Issue by Darker** [mkozjak/node-telnet-client#121]
This should work but does not:

```
try {
    await connection.connect(options);
}
catch(e) {
    console.error("Connection failed!",e.stack);
}
```

Instead, the Node.js process is terminated with this error:

```
events.js:173
    throw err; // Unhandled 'error' event
    ^

Error [ERR_UNHANDLED_ERROR]: Unhandled error. (Cannot connect)
    at Telnet.emit (events.js:171:17)
    at Socket.socket.setTimeout (D:\web\node-smart-tv\SmartTV\SmartTV\node_modules\telnet-client\lib\index.js:76:16)
    at Object.onceWrapper (events.js:273:13)
    at Socket.emit (events.js:182:13)
    at Socket._onTimeout (net.js:444:8)
    at ontimeout (timers.js:427:11)
    at tryOnTimeout (timers.js:289:5)
    at listOnTimeout (timers.js:252:5)
    at Timer.processTimers (timers.js:212:10)
```
If you assign `error` event callback, the the Node.js process does not crash, but the `connect()` promise never rejects. This is a major issue which prevents any Promise-based usage of this otherwise great library.

**Comment by mkozjak** [comment]
Hey, @Darker, thanks for reporting. Looking into this.

**Comment by Darker** [comment]
@mkozjak I went throught the code and essentially there are missing `reject` calls in the Promise that does the main connecting. It does emit the error event, but reject is never called and so the promise hangs forever.

**Comment by mkozjak** [comment]
Fixed in 0.16.4.
